pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -243,6 +258,33 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 	c.serviceLister = kubeClient.KubeInformer().Core().V1().Services().Lister()
 	registerHandlers(c.serviceInformer, c.queue, "Services", c.onServiceEvent, nil)
 
+	if enableEventing.Get() {
+		// To get events posted by other Istiod instances, in order to sync and support debug tools.
+		// This will also allow us to get readiness/status when VM health check is done, and possibly get faster
+		// updates about endpoints.
+		c.eventsInformer = kubeClient.KubeInformer().Core().V1().Events().Informer()
+		c.eventsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if ev, ok := obj.(*corev1.Event); ok {
+					c.onEvent(ev, "add")
+				}
+			},
+			UpdateFunc: func(old, obj interface{}) {
+				if ev, ok := obj.(*corev1.Event); ok {
+					c.onEvent(ev, "update")
+				}
+			},
+		})
+
+		c.eventBroadcaster = record.NewBroadcaster()
+		// TODO: add revision - from MeshConfig ( after it is added !)
```

Comment:
we have revision, its just not in mesh config

Reply:
Yes, I didn't want to bother with passing it - we should just add it to MeshConfig and use mesh config.